### DISCUSSION TITLE
Extra Test data

### DISCUSSION THREAD

**Discussion by Jeremy-QA** [seleniumbase/SeleniumBase#2810] [Q&A] [answered]
Hello @mdmintz I tried to create my own customized parameter name, similar to the extra test data accessed with 'self.var2' and '--var2' in the test parameters. However, whenever I update SeleniumBase, my customized extra test data with the parameter is removed.

How can I update SeleniumBase without my customized parameter being removed? Thank you very much!

**Comment by mdmintz** [comment]
Duplicate of https://github.com/seleniumbase/SeleniumBase/discussions/2110#discussioncomment-6996500

Use ``--variables=DICT`` to pass in custom variables.

Eg: ``pytest --variables='{"MY_VAR":"abc", "VAR123":123}'``

```bash
[1] > HOME/SeleniumBase/examples/test_null.py(7)
   .
   6         def test_null(self):
   7  ->         pass
(Pdb+) self.variables
{'MY_VAR': 'abc', 'VAR123': 123}
(Pdb+) self.variables["MY_VAR"]
'abc'
```

**Reply by Jeremy-QA** [reply]
hello @mdmintz  Could you assist me in implementing that based on my approach using self.var1?


```
def test(self):
    device_list = self.var1

    if device_list == "android":
        print("The device is:", device_list)
    elif device_list == "ios":
        print("The device is:", device_list)
        
        
```

pytest .\practice.py --var1=ios -s
The device is: ios

**Reply by Jeremy-QA** [reply, marked as answer]
i think i got it @mdmintz  and its working :)

    
```
def test(self):
        device_list = self.variables.get("device")
        
        if device_list == "android":
            print("The device is:", device_list)
        elif device_list == "ios":
            print("The device is:", device_list)
        else:
            print("not found")
```


pytest .\practice.py --variables="{'device':'ios'}" -s
The device is: ios
